site/docs/glossary.md

```diff
@@ -0,0 +1,537 @@
+directory to store outputs. Prepared during the loading phase by planting a
+"symlink forest" of the directories representing transitive closure of packages
+depended on by the target(s) to build. Accessible with `bazel info
+execution_root` on the command line.
+
+#### File
+
+See [*Artifact*](#artifact).
+
+#### Hermeticity
+
+The property of a build or test where there are no external influences on the
+build or test. This helps to enable deterministic and correct results. This
+usually means disallowing network access to actions, only allowing access to
+declared input [*artifacts*](#artifact), making timestamps and timezones fixed,
+restricting access/resetting environment variables, and using fixed seeds for
+random number generators.
+
+#### Incremental build
+
+A build that reuses the results of earlier builds to reduce build time and
+resource needs. Bazel aims to guarantee correct results for incremental builds.
+This is the opposite of a clean build.
+
+<!-- TODO: #### Install base -->
+
+#### Label
+
+An identifier for a target. A fully qualified label such as
+`//path/to/package:target` consists of `//` to mark the workspace root
+directory, path/to/package as the directory that contains the [*BUILD
+file*](#build-file) declaring the target, and :target as the name of the target
+declared in the aforementioned BUILD file. May also be prefixed with
+`@my_repository//<..>` to indicate that the target is declared in an *external
+repository* named `my_repository`.
+
+#### Loading phase
+
+The first phase of a build. WORKSPACE, BUILD, and .bzl files are parsed to
+create *packages*. Macros and certain functions like glob() are evaluated in
+this phase. Interleaved with the second phase of the build, the *analysis
+phase*, to build up a *target dependency graph*.
+
+#### Macro
+
+A mechanism to compose multiple rule target declarations together under a single
+Starlark function. Enables reusing common rule declaration patterns across BUILD
+files. Expanded to the underlying rule target declarations during the *loading
+phase*.
+
+#### Mnemonic
+
+A short, human-readable string to quickly understand what an *action* is doing.
+Can be used as identifiers for *spawn strategy* selections. For example: Javac,
+CppCompile, AndroidManifestMerger.
+
+#### Native rules
+
+*Rules* that are implemented in Java and built into Bazel. Such rules appear in
+*.bzl files* as functions in the native module (for example, `native.cc_library`
+or `native.java_library`). User-defined rules (non-native) are created using
+Starlark.
+
+#### Output base
+
+A unique directory specific to a workspace for containing Bazel output files to
+separate outputs from the source tree. Located in the *output user root*.
+
+#### Output groups
+
+A group of files that is expected to be built when Bazel finishes building a
+target. Rules put their usual outputs in the "default output group" (e.g the
```

Comment:
As above, it helps users out to set apart file types like `.jar`, `.a` and `.so` with backticks as you have here with `cc_library`.